search/templates/cosmos/searchresults.html

```diff
@@ -31,57 +62,173 @@ <h6>
                                                 class="st-custom-button fa fa-google-plus-square"></div></a>
     </span>
 
+    <div class="tab">
+      <button class="tablinks" onclick="display(event, 'All')">All</button>
+      <button class="tablinks" onclick="display(event, 'Code')">Code</button>
+      <button class="tablinks" onclick="display(event, 'External')">External</button>
+    </div>
+
+    <script>
+      function display(evt, Name) {
+          var i, tabcontent, tablinks;
+          tabcontent = document.getElementsByClassName("tabcontent");
+          for (i = 0; i < tabcontent.length; i++) {
+              tabcontent[i].style.display = "none";
+          }
+          tablinks = document.getElementsByClassName("tablinks");
+          for (i = 0; i < tablinks.length; i++) {
+              tablinks[i].className = tablinks[i].className.replace(" active", "");
+          }
+          document.getElementById(Name).style.display = "block";
+          evt.currentTarget.className += " active";
+      }
+</script>
     </h6>
-    <hr>
     {% if result_val or not result%}
         {% include 'cosmos/calculator.html' %}
     {% endif %}
+    <div id="All" class="tabcontent">
```

Comment:
You could remove this 'All' part, and add javascript to determine which part or whole need to be displayed.
In addition, all of result should be displayed by default should be better than.